### PR TITLE
Add a  wait and retry strategy when etcdwatch fails.

### PR DESCRIPTION
**Pull request by jlevesy**

### CODE REVIEW

File: etcdexpose/etcdwatcher.go [hunk 1]
```diff
@@ -8,25 +8,37 @@ import (
...
+		eventChan,
+		e.stopChan)
+	e.running = false
+	errorChan <- err
+	return
+}
+
+func (e *EtcdWatcher) Stop() {
+	if e.running {
+		e.stopChan <- true
```
Comment (jlevesy) [inline etcdexpose/etcdwatcher.go]: Actually this push is blocking if watch has already returned, I had to protect it from this case.

Comment (jlevesy) [inline etcdexpose/etcdwatcher.go]: This is clearly not a solution.

**Comment by AlexisMontagne** [comment]
Why not just lookup the error and only ignore then?

**Comment by jlevesy** [comment]
Huu, maybe to retry the watch?  If we simply ignore the error, we will not retry to watch the key ?

**Comment by jlevesy** [comment]
client.Watch() methods returns on error

**Comment by AlexisMontagne** [comment]
By ignoring i just mean  put  the `Watch` in a `for` loop and lookup the error of the command and only continue if it's not  critical

**Comment by jlevesy** [comment]
:+1:  Smart.
However If there is a timewatcher active, we will continue to try to update watched key, even if etcd is not reachable.  This is why I thought to implement this at runner level.
As you wish :)

**Comment by AlexisMontagne** [comment]
LGTM

**Pull request merged**